crowbar_framework/app/assets/javascripts/installer.js

```diff
@@ -0,0 +1,45 @@
+$(document).ready(function() {
+  function statusCheck() {
+    $.ajax({
+      type: "GET",
+      dataType: "json",
+      url: "/installer/status.json",
+      success: function(data)
+      {
+        var failed = data.failed;
+        data = data.steps;
+        var mostRecent = $("li." + data[data.length-1]);
+        var mostRecentIcon = mostRecent.children();
+
+        $.each(data, function (index, value) {
+          $("li." + value).removeClass("list-group-item-success");
+          $("li." + value).children("span").removeClass("fa-hourglass-o fa-circle-o-notch fa-spin");
+          $("li." + value).children("span").addClass("fa-check");
+        });
+
+        mostRecentIcon.removeClass(function (index, css) {
+          match = css.match(/^fa-/g);
+          if (match !== null) {
+            return match.join("");
+          }
+        });
+        mostRecentIcon.addClass("fa-check");
+
+        if (failed) {
+          mostRecent.next("li").addClass("list-group-item-danger");
+          mostRecent.next("li").children("span").removeClass("fa-hourglass-o fa-circle-o-notch fa-spin");
+          mostRecent.next("li").children("span").addClass("fa-times");
+        } else {
+          mostRecent.next("li").addClass("list-group-item-success");
+          mostRecent.next("li").children("span").removeClass("fa-hourglass-o");
+          mostRecent.next("li").children("span").addClass("fa-circle-o-notch fa-spin");
+        }
+      },
+      error: function(data) {
+      }
+    });
+  }
+  // call statusCheck to not wait for setInterval to kick in
+  statusCheck();
+  setInterval(statusCheck, 3000);
```

Comment:
Overall you need to match for some specific element, else the installer ajax will run forever, even after the installation.